src/Kriti/Parser/Token.hs

```diff
@@ -151,5 +155,56 @@ instance Located ValueExt where
 
 instance Located Accessor where
   locate = \case
-    Obj s _ -> s
+    Obj s _ _ -> s
     Arr s _ -> s
+
+instance Pretty ValueExt where
+  pretty = \case
+    Object _ km ->
+      let p :: (T.Text, ValueExt) -> Doc a
+          p (k, v) = pretty k <> colon <+> pretty v <> comma
+       in braces $ braces $ foldMap p $ Compat.toList km
+    Array _ vec -> pretty $ V.toList vec
+    String _ txt -> dquotes (pretty txt)
+    Number _ sci -> pretty $ show sci
+    Boolean _ b -> pretty b
+    Null _ -> "null"
+    StringTem _ vec ->
+      dquotes $
+        flip foldMap vec $ \case
```

Comment:
This isn't even a suggestion, more just letting you know of an alternative pattern:

With `-XBlockArguments` you can write `vec & foldMap \case [...]` which (IMO) reads a bit more nicely (almost like method call syntax).

Reply:
oh I like that and actually have BlockArguments enabled. I always forget to use them